src/track/beatgrid.cpp

```diff
@@ -67,27 +67,26 @@ void BeatGrid::setGrid(double dBpm, double dFirstBeatSample) {
     m_dBeatLength = (60.0 * m_iSampleRate / dBpm) * kFrameSize;
 }
 
-QByteArray* BeatGrid::toByteArray() const {
+std::unique_ptr<QByteArray> BeatGrid::toByteArray() const {
     QMutexLocker locker(&m_mutex);
     std::string output;
     m_grid.SerializeToString(&output);
-    QByteArray* pByteArray = new QByteArray(output.data(), output.length());
-    // Caller is responsible for delete
-    return pByteArray;
+    return std::make_unique<QByteArray>(output.data(), output.length());
```

Comment:
I think we can improve this even more. 
This function is only used here 
https://github.com/mixxxdj/mixxx/blob/87ef849b94e6080b307fff33793567682c0687a2/src/library/dao/trackdao.cpp#L1605
To finally create a temorary QVariant. 

How about
```
QVariant BeatGrid::toVariant() const {
     QMutexLocker locker(&m_mutex);
     std::string output;
     m_grid.SerializeToString(&output);
     return QVariant(output.c_str);
}
```